wgpu-hal/src/vulkan/instance.rs

```diff
@@ -114,6 +114,125 @@ impl super::Swapchain {
 }
 
 impl super::Instance {
+    pub fn required_extensions(
+        entry: &ash::Entry,
+        driver_api_version: u32,
+        with_debug: bool,
```

Comment:
Let's just pass `hal::InstanceFlags` here. Since the external user would pass in a `hal::Instance`, they already depend on wgpu-hal